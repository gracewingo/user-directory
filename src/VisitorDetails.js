import React from 'react';

const VisitorDetails = ({ profileData }) => (
	<div className="preview-pane">
		<h1>Visitor Details</h1>
		<div id="profile">
			<div>
				Name: {profileData[0].first_name} {profileData[0].last_name}
			</div>
			<div>City: {profileData[0].city}</div>
			<div>State: {profileData[0].state}</div>
			<div>Safe email: {profileData[0].safe_email}</div>
			<div>Email: {profileData[0].email}</div>
		</div>
	</div>
);

export default VisitorDetails;
