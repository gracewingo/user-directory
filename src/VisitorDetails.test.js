import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import VisitorDetails from './VisitorDetails';
import { create } from 'react-test-renderer';

configure({ adapter: new Adapter() });

const data = [ { id: 21, name: 'Bob' } ];

describe('VisitorDetails component', () => {
	it('Matches the snapshot', () => {
		const visDetails = create(<VisitorDetails profileData={data} />);
		expect(visDetails.toJSON()).toMatchSnapshot();
	});

	it('shallow render', () => {
		shallow(<VisitorDetails profileData={data} />);
	});
});
