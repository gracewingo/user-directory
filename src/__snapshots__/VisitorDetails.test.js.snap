// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisitorDetails component Matches the snapshot 1`] = `
<div
  className="preview-pane"
>
  <h1>
    Visitor Details
  </h1>
  <div
    id="profile"
  >
    <div>
      Name: 
       
    </div>
    <div>
      City: 
    </div>
    <div>
      State: 
    </div>
    <div>
      Safe email: 
    </div>
    <div>
      Email: 
    </div>
  </div>
</div>
`;
